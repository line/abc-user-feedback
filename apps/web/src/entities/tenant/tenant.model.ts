/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { create } from 'zustand';

import { client } from '@/shared';

import type { Tenant } from './tenant.type';

type State = { tenant: Tenant | null };

type Action = {
  setTenant: (tenant: Tenant) => void;
  refetchTenant: () => Promise<void>;
};

export const useTenantStore = create<State & Action>((set) => ({
  tenant: null,
  setTenant: (tenant) => set({ tenant }),
  refetchTenant: async () => {
    const { data } = await client.get({ path: '/api/admin/tenants' });
    set({ tenant: data });
  },
}));
