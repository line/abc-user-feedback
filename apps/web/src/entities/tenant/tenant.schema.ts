/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import { z } from 'zod';

export const tenantSchema = z.object({
  id: z.number(),
  siteName: z.string(),
  description: z.string().nullable(),
  useEmail: z.boolean(),
  useOAuth: z.boolean(),
  isPrivate: z.boolean(),
  isRestrictDomain: z.boolean(),
  allowDomains: z
    .array(z.string().refine((v) => /[a-z]+\.[a-z]{2,3}/.test(v)))
    .nullable(),
  useEmailVerification: z.boolean(),
  oauthConfig: z
    .object({
      clientId: z.string(),
      emailKey: z.string(),
      scopeString: z.string(),
      clientSecret: z.string(),
      authCodeRequestURL: z.string(),
      accessTokenRequestURL: z.string(),
      userProfileRequestURL: z.string(),
    })
    .nullable(),
});

export const tenantInfoSchema = tenantSchema.pick({
  id: true,
  siteName: true,
  description: true,
});
export const authInfoScema = tenantSchema.pick({
  isPrivate: true,
  isRestrictDomain: true,
  allowDomains: true,
  useOAuth: true,
  useEmail: true,
  oauthConfig: true,
});
