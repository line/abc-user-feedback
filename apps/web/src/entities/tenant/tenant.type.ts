/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { z } from 'zod';

const oauthConfigSchema = z.object({
  oauthUse: z.boolean(),
  clientId: z.string(),
  clientSecret: z.string(),
  authCodeRequestURL: z.string(),
  scopeString: z.string(),
  accessTokenRequestURL: z.string(),
  userProfileRequestURL: z.string(),
  emailKey: z.string(),
});

const tenantSchema = z.object({
  id: z.number(),
  siteName: z.string(),
  description: z.string().nullable(),
  useEmail: z.boolean(),
  useOAuth: z.boolean(),
  isPrivate: z.boolean(),
  isRestrictDomain: z.boolean(),
  allowDomains: z.array(z.string()),
  useEmailVerification: z.boolean(),
  oauthConfig: oauthConfigSchema.nullable(),
});

export type Tenant = z.infer<typeof tenantSchema>;

export type OAuthConfig = z.infer<typeof oauthConfigSchema>;
