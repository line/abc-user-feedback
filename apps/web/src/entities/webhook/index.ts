export * from './ui';
export * from './webhook.type';
