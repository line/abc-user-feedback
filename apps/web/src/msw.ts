/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import { http, HttpResponse } from 'msw';
import { setupServer } from 'msw/node';

import { signInWithOAuthMockHandlers } from './features/auth/sign-in-with-oauth/__mocks__/sign-in-with-oauth.mock-handler';
import type {
  OAIMethodPathKeys,
  OAIMethods,
  OAIPathParameters,
} from './shared';
import { convertToColonPath, getRequestUrl } from './shared';

export const server = setupServer(...signInWithOAuthMockHandlers);

export const simpleMockHttp = <
  TMethod extends OAIMethods,
  TPath extends OAIMethodPathKeys<TMethod>,
>({
  method,
  path,
  status = 200,
  params,
  data = {},
}: {
  method: TMethod;
  path: TPath;
  status?: 200 | 201 | 204 | 400 | 401 | 403 | 404 | 500;
  params?: OAIPathParameters<TPath, TMethod>;
  data?: Record<string, unknown>;
}) =>
  server.use(
    http[method](`${convertToColonPath(getRequestUrl(path, params))}`, () =>
      HttpResponse.json(data, { status }),
    ),
  );
