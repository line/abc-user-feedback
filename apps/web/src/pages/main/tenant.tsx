/**
 * Copyright 2025 LY Corporation
 *
 * LY Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import type { GetStaticProps } from 'next';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';
import { parseAsString, useQueryState } from 'nuqs';

import { Icon, Menu, MenuItem } from '@ufb/react';

import type { NextPageWithLayout } from '@/shared';
import SideMenuLayout from '@/shared/ui/side-menu-layout.ui';
import { Layout } from '@/widgets/layout';
import {
  LoginSetting,
  TenantInfoSetting,
  UserManagementSetting,
} from '@/widgets/setting-menu';

import serverSideTranslations from '@/server-side-translations';

const TenantPage: NextPageWithLayout = () => {
  const { t } = useTranslation();
  const router = useRouter();
  const [currentMenu] = useQueryState<string>(
    'menu',
    parseAsString.withDefault('tenant'),
  );

  return (
    <SideMenuLayout
      sideMenu={
        <Menu
          type="single"
          orientation="vertical"
          className="w-full p-0"
          value={currentMenu}
          onValueChange={(value) =>
            router.push({ pathname: router.pathname, query: { menu: value } })
          }
        >
          <MenuItem value="tenant">
            <Icon name="RiInformation2Line" />
            {t('tenant-setting-menu.tenant-info')}
          </MenuItem>
          <MenuItem value="login">
            <Icon name="RiUser2Line" />
            {t('tenant-setting-menu.sign-up-mgmt')}
          </MenuItem>
          <MenuItem value="user">
            <Icon name="RiShieldKeyholeLine" />
            {t('tenant-setting-menu.user-mgmt')}
          </MenuItem>
        </Menu>
      }
    >
      {currentMenu === 'tenant' && <TenantInfoSetting />}
      {currentMenu === 'login' && <LoginSetting />}
      {currentMenu === 'user' && <UserManagementSetting />}
    </SideMenuLayout>
  );
};

TenantPage.getLayout = (page) => {
  return <Layout title="Tenant">{page}</Layout>;
};

export const getStaticProps: GetStaticProps = async ({ locale }) => {
  return {
    props: {
      ...(await serverSideTranslations(locale)),
    },
  };
};

export default TenantPage;
