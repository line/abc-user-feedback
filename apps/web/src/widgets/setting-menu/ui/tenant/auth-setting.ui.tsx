/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import { useEffect, useState } from 'react';
import { zodResolver } from '@hookform/resolvers/zod';
import { useTranslation } from 'next-i18next';
import { FormProvider, useForm } from 'react-hook-form';

import { Badge, Input, toast } from '@ufb/ui';

import { DescriptionTooltip, RadioGroup, useOAIMutation } from '@/shared';
import type { AuthInfo } from '@/entities/tenant';
import {
  authInfoScema,
  OAuthConfigForm,
  useTenantStore,
} from '@/entities/tenant';

import SettingMenuTemplate from '../setting-menu-template';

type DomainStateType = {
  isSubmitted: boolean;
  isValid: boolean;
  hint?: string;
};

interface IProps {}

const AuthSetting: React.FC<IProps> = () => {
  const { t } = useTranslation();

  const { tenant, refetchTenant } = useTenantStore();

  const methods = useForm<AuthInfo>({
    resolver: zodResolver(authInfoScema),
  });

  const { reset, handleSubmit, watch, setValue, formState } = methods;

  const [domainState, setDomainState] = useState<DomainStateType>({
    isSubmitted: false,
    isValid: false,
  });

  const [currentDomain, setCurrentDomain] = useState('');

  const { mutate, isPending } = useOAIMutation({
    method: 'put',
    path: '/api/admin/tenants',
    queryOptions: {
      async onSuccess() {
        await refetchTenant();
        toast.positive({ title: t('toast.save') });
        setDomainState({ isSubmitted: false, isValid: false });
      },
      onError(error) {
        toast.negative({ title: error?.message ?? 'Error' });
      },
    },
  });

  useEffect(() => {
    if (!tenant) return;
    reset(tenant);
  }, [tenant]);

  const onClickDelete = (targetIndex: number) => () => {
    setValue(
      'allowDomains',
      (watch('allowDomains') ?? []).filter((_, i) => i !== targetIndex),
      { shouldDirty: true },
    );
  };

  const onClickAdd = () => {
    setDomainState({ isSubmitted: true, isValid: true });
    if (!currentDomain || currentDomain.length < 1) {
      setDomainState((prev) => ({
        ...prev,
        isValid: false,
        hint: t('hint.required'),
      }));
      return;
    }

    const domain = currentDomain.slice(currentDomain.startsWith('@') ? 1 : 0);

    if (!/[a-z]+\.[a-z]{2,3}/.test(domain)) {
      setDomainState((prev) => ({
        ...prev,
        isValid: false,
        hint: t('hint.invalid-domain'),
      }));
      return;
    }

    setValue(`allowDomains.${(watch('allowDomains') ?? []).length}`, domain, {
      shouldDirty: true,
    });

    setCurrentDomain('');
    setDomainState((prev) => ({ ...prev, isValid: true }));
  };

  const onSubmit = (input: AuthInfo) => {
    if (!tenant) return;
    if (input.isRestrictDomain) {
      input.allowDomains = (input.allowDomains ?? []).filter(
        (v) => v && v.length > 0,
      );
    } else {
      input.allowDomains = [];
    }
    mutate({ ...tenant, ...input });
  };

  return (
    <SettingMenuTemplate
      title={t('tenant-setting-menu.sign-up-mgmt')}
      actionBtn={{
        children: t('button.save'),
        onClick: handleSubmit(onSubmit),
        form: 'form',
        type: 'submit',
        disabled: !formState.isDirty || isPending,
      }}
    >
      <FormProvider {...methods}>
        <form id="form" className="flex flex-col gap-6">
          <div className="flex flex-col gap-4">
            <p className="input-label flex items-center gap-1">Login Method</p>
            <RadioGroup
              name="signUpMethod"
              radios={[
                {
                  label: t('main.setting.sign-up-mgmt.ufb-login'),
                  checked: watch('useEmail') && !watch('useOAuth'),
                  onChecked: (isChecked) => {
                    if (!isChecked) return;
                    setValue('useEmail', true, { shouldDirty: true });
                    setValue('useOAuth', false, { shouldDirty: true });
                  },
                },
                {
                  label: t('main.setting.sign-up-mgmt.oauth-login'),
                  checked: !watch('useEmail') && watch('useOAuth'),
                  onChecked: (isChecked) => {
                    if (!isChecked) return;
                    setValue('useEmail', false, { shouldDirty: true });
                    setValue('useOAuth', true, { shouldDirty: true });
                  },
                },
                {
                  label: t('main.setting.sign-up-mgmt.all-login'),
                  checked: watch('useEmail') && watch('useOAuth'),
                  onChecked: (isChecked) => {
                    if (!isChecked) return;
                    setValue('useEmail', true, { shouldDirty: true });
                    setValue('useOAuth', true, { shouldDirty: true });
                  },
                },
              ]}
            />
            {watch('useOAuth') && <OAuthConfigForm />}
          </div>
          <div className="flex flex-col gap-4">
            <p className="input-label flex items-center gap-1">
              Sign Up Method
            </p>
            <RadioGroup
              name="isPrivate"
              radios={[
                {
                  label: t('main.setting.sign-up-mgmt.possibility'),
                  checked: !watch('isPrivate'),
                  onChecked: (isChecked) => {
                    if (!isChecked) return;
                    setValue('isPrivate', false, { shouldDirty: true });
                  },
                },
                {
                  label: t('main.setting.sign-up-mgmt.impossibility'),
                  checked: watch('isPrivate'),
                  onChecked: (isChecked) => {
                    if (!isChecked) return;
                    setValue('isPrivate', true, { shouldDirty: true });
                  },
                },
              ]}
            />
          </div>
          <div className="flex flex-col gap-4">
            <p className="input-label flex items-center gap-1">
              Email domain WhiteList
              <DescriptionTooltip
                description={t(
                  'main.setting.sign-up-mgmt.domain-restriction-tooltip',
                )}
              />
            </p>
            <RadioGroup
              name="isRestrictDomain"
              radios={[
                {
                  label: t('main.setting.sign-up-mgmt.no-domain-restriction'),
                  checked: !watch('isRestrictDomain'),
                  onChecked: () =>
                    setValue('isRestrictDomain', false, { shouldDirty: true }),
                },
                {
                  label: t('main.setting.sign-up-mgmt.domain-restriction'),
                  checked: watch('isRestrictDomain'),
                  onChecked: () =>
                    setValue('isRestrictDomain', true, { shouldDirty: true }),
                },
              ]}
            />
            {watch('isRestrictDomain') && (
              <div className="flex flex-col gap-2">
                <Input
                  value={currentDomain}
                  placeholder="@example.com"
                  onChange={(e) => {
                    if (domainState.isSubmitted && domainState.isValid) {
                      setDomainState({ isSubmitted: false, isValid: false });
                    }
                    setCurrentDomain(e.target.value);
                  }}
                  onKeyDown={(e) => {
                    if (e.key === 'Enter') {
                      e.preventDefault();
                      onClickAdd();
                    }
                  }}
                  rightChildren={
                    <button
                      type="button"
                      className="btn btn-xs btn-rounded btn-primary"
                      onClick={onClickAdd}
                    >
                      {t('button.register')}
                    </button>
                  }
                  {...domainState}
                />
                <div className="flex flex-row flex-wrap gap-2">
                  {(watch('allowDomains') ?? []).map((domain, i) => (
                    <Badge
                      key={i}
                      type="secondary"
                      right={{ iconName: 'Close', onClick: onClickDelete(i) }}
                    >
                      @{domain}
                    </Badge>
                  ))}
                </div>
              </div>
            )}
          </div>
        </form>
      </FormProvider>
    </SettingMenuTemplate>
  );
};

export default AuthSetting;
