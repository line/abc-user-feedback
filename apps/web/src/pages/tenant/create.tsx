/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */
import type { GetStaticProps } from 'next';
import { useRouter } from 'next/router';
import { zodResolver } from '@hookform/resolvers/zod';
import { useTranslation } from 'next-i18next';
import { serverSideTranslations } from 'next-i18next/serverSideTranslations';
import { useForm } from 'react-hook-form';
import { z } from 'zod';

import { toast } from '@ufb/ui';

import AuthTemplate from '@/components/templates/AuthTemplate';
import { DEFAULT_LOCALE } from '@/constants/i18n';
import { Path } from '@/constants/path';
import { useTenant } from '@/contexts/tenant.context';
import { useOAIMutation } from '@/hooks';
import type { NextPageWithLayout } from '../_app';

interface IForm {
  siteName: string;
}

const schema: Zod.ZodType<IForm> = z.object({
  siteName: z.string(),
});

const defaultValues: IForm = {
  siteName: '',
};

const CreatePage: NextPageWithLayout = () => {
  const { t } = useTranslation();

  const router = useRouter();
  const { refetch } = useTenant();

  const { register, handleSubmit } = useForm({
    resolver: zodResolver(schema),
    defaultValues,
  });

  const { mutate, isPending } = useOAIMutation({
    method: 'post',
    path: '/api/admin/tenants',
    queryOptions: {
      async onSuccess() {
        toast.positive({ title: 'Success' });
        toast.positive({
          title: 'create Default Super User',
          description: 'email: user@feedback.com \n password: 12345678',
        });
        router.push(Path.SIGN_IN);
        refetch();
      },
      onError(error) {
        toast.negative({ title: error?.message ?? 'Error' });
      },
    },
  });
  const onSubmit = (data: IForm) => mutate(data);

  return (
    <AuthTemplate>
      <div className="flex h-screen w-screen items-center justify-center">
        <div className="w-full max-w-[400px] rounded border p-4 shadow">
          <form
            onSubmit={handleSubmit(onSubmit)}
            className="flex flex-col gap-4"
          >
            <h1 className="font-20-bold">{t('tenant.create.title')}</h1>
            <label>
              <span>{t('tenant.create.site-name')}</span>
              <input className="input" type="text" {...register('siteName')} />
            </label>

            <button
              className="btn btn-primary"
              type="submit"
              disabled={isPending}
            >
              {t('button.setting')}
            </button>
          </form>
        </div>
      </div>
    </AuthTemplate>
  );
};

export const getStaticProps: GetStaticProps = async ({ locale }) => {
  return {
    props: {
      ...(await serverSideTranslations(locale ?? DEFAULT_LOCALE)),
    },
  };
};

export default CreatePage;
