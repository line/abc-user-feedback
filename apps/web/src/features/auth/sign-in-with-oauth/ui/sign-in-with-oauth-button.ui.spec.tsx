/**
 * Copyright 2023 LINE Corporation
 *
 * LINE Corporation licenses this file to you under the Apache License,
 * version 2.0 (the "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

import { faker } from '@faker-js/faker';
import userEvent from '@testing-library/user-event';
import mockRouter from 'next-router-mock';

import type { Tenant } from '@/entities/tenant';
import { useTenantStore } from '@/entities/tenant';

import SignInWithOAuthButton from './sign-in-with-oauth-button.ui';

import { simpleMockHttp } from '@/msw';
import { render, screen, waitFor } from '@/test-utils';

describe('SignInWithOAuthButton', () => {
  test('match snapshot', () => {
    const component = render(<SignInWithOAuthButton />);
    expect(component.container).toMatchSnapshot();
  });
  test('loginUrl', async () => {
    useTenantStore.setState({ tenant: { useOAuth: true } as Tenant });
    const pathname = `/${Array.from({
      length: faker.number.int({ min: 1, max: 5 }),
    })
      .map(() => faker.string.alphanumeric({ length: { min: 1, max: 5 } }))
      .join('/')}`;

    simpleMockHttp({
      method: 'get',
      path: '/api/admin/auth/signIn/oauth/loginURL',
      status: 200,
      data: { url: pathname },
    });

    render(<SignInWithOAuthButton />);

    await waitFor(() => expect(screen.getByRole('button')).not.toBeDisabled());
    await userEvent.click(screen.getByRole('button'));

    expect(mockRouter).toMatchObject({ asPath: pathname });
  });
});
