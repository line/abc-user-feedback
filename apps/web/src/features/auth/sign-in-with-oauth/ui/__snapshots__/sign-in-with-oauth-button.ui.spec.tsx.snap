// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInWithOAuthButton match snapshot 1`] = `
<div>
  <button
    class="btn btn-lg btn-blue"
    disabled=""
    type="button"
  >
    OAuth2.0 
    button.sign-in
  </button>
  <div
    style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
  />
</div>
`;
